homeassistant/components/flo/device.py

```diff
@@ -0,0 +1,174 @@
+"""Flo device object."""
+from datetime import datetime, timedelta
+import logging
+from typing import Any, Callable, Dict, List
+
+from aioflo.api import API
+
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import SIGNAL_DEVICE_UPDATED, SIGNAL_WATER_CONSUMPTION_UPDATED
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class FloDevice:
+    """Flo device object."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        device_information: Dict[str, Any],
+        api_client: API,
+    ):
+        """Initialize the device."""
+        self.hass: HomeAssistantType = hass
+        self.api_client: API = api_client
+        self.unsubs: List[Callable] = []
+        self._flo_device_id: str = device_information["id"]
+        self._flo_location_id: str = device_information["location"]["id"]
+        self._mac_address: str = device_information["macAddress"]
+        self._manufacturer: str = "Flo by Moen"
+        self._model: str = device_information["deviceModel"]
+        self._device_type: str = device_information["deviceType"]
+        self._serial_number: str = device_information["serialNumber"]
```

Comment:
Instead of all this, why not use `@property` ? that way the information is updated all at once. Or, since it's just for device info, just have a `device_info` property?

Reply:
I personally prefer the individual properties here that way everything accessing this object doesn't have to muck w/ a dict but I can change it if you want. Just lmk

Reply:
well was mainly thinking that if it's only used for `Entity.device_info`, let's just have that be returned.

Reply:
it's not. The fields are used in states etc. device info only uses a couple of these fields.